darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1429,6 +1419,105 @@ def _load_from_checkpoint(self, file_path, **kwargs):
+        that need to be re-trained or fine-tuned with different optimizer or learning rate scheduler.
+
+        To resume an interrupted training, please consider using :meth:`load_from_checkpoint()
+        <TorchForecastingModel.load_from_checkpoint()>` which also reload the trainer, optimizer and
+        learning rate scheduler states.
+
+        For manually saved model, consider using :meth:`load() <TorchForecastingModel.load()>` instead.
+
+        Parameters
+        ----------
+        model_name
+            The name of the model (used to retrieve the checkpoints folder's name). Default: ``self.model_name``.
+        work_dir
+            Working directory (containing the checkpoints folder). Defaults to current working directory.
+        file_name
+            The name of the checkpoint file. If not specified, use the most recent one.
+        best
+            If set, will retrieve the best model (according to validation loss) instead of the most recent one. Only
+            is ignored when ``file_name`` is given. Default: ``True``.
+        strict
+            If set, strictly enforce that the keys in state_dict match the keys returned by this module’s state_dict().
+            Default: ``True``.
+            For more information, read the `official documentation <https://pytorch.org/docs/stable/generated/torch.
+            nn.Module.html?highlight=load_state_dict#torch.nn.Module.load_state_dict>`_.
+        **kwargs
+            Additional kwargs for PyTorch's :func:`load` method, such as ``map_location`` to load the model onto a
+            different device than the one from which it was saved.
+            For more information, read the `official documentation <https://pytorch.org/docs/stable/generated/
+            torch.load.html>`_.
+        """
+        raise_if(
+            "weights_only" in kwargs.keys() and kwargs["weights_only"],
+            "Passing `weights_only=True` to `torch.load` will disrupt this"
+            " method sanity checks.",
+            logger,
+        )
+
+        # use the name of the model being loaded with the saved weights
+        if model_name is None:
+            model_name = self.model_name
+
+        if work_dir is None:
+            work_dir = os.path.join(os.getcwd(), DEFAULT_DARTS_FOLDER)
+
+        # load PyTorch LightningModule from checkpoint
+        # if file_name is None, find most recent file in savepath that is a checkpoint
+        if file_name is None:
+            file_name = _get_checkpoint_fname(work_dir, model_name, best=best)
+
+        checkpoint_dir = _get_checkpoint_folder(work_dir, model_name)
+
+        ckpt = torch.load(os.path.join(checkpoint_dir, file_name), **kwargs)
+        ckpt_hyper_params = ckpt["hyper_parameters"]
+
+        # verify that the arguments passed to the constructor match those of the checkpoint
+        for param_key, param_value in self.model_params.items():
```

Comment:
If we hard set `strict=True` we could let PyTorch handle any discrepancies later on when calling 
```
self.model.load_state_dict(ckpt["state_dict"], strict=True)
```

Reply:
This is what I did initially but the error message are not informative at all, making it quite difficult for the user to realize that the problem comes from the definition of the model into which the weights are loaded.

Reply:
True, but on the other hand this requires all TorchForecastingModels and their corresponding PLForecastingModules to share the same model parameter names, which is not the case as you mention (and might be difficult to enforce in some cases).

So the torch error can still be raised, or maybe I'm missing something :)

Reply:
Indeed, the torch error can still be raised if the discrepancy is in one of these parameter that do not have the same name in these two objects.

We could eventually have a dict in each model that that tries to map the name of the parameters in order to be able to run this sanity check thoroughly? Or try to catch torch error if `load_state_dict` fails and raise a meaningful message to the user, indicating that the weights mismatch can be caused by invalid parameters (or by a change of convention in torch...)?